src/System.Text.Json/tests/Serialization/JsonElementTests.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Linq;
+using Xunit;
+
+namespace System.Text.Json.Serialization.Tests
+{
+    public class JsonElementTests
+    {
+        [Fact]
+        public void SerializeJsonElement()
+        {
+            var obj = JsonSerializer.Parse<JsonElementClass>(JsonElementClass.s_json);
```

Comment:
nit: avoid var here.

Reply:
That's within our guidelines.

Reply:
How so? What is the type of obj? Isn't it being `JsonElementClass` an implementation detail of this particular `Parse` method (which as a reviewer I may not know ahead of time).

Reply:
Presumption is that you would and that this is test code. I'll change them.

Reply:
The rule is a bit foofy "only when it's obvious".  The easiest interpretation is only allowed for `new`, `as`, and "hard" casts.  While the generic parameter here sort of makes the return type obvious, it could also have been a base-class qualifier on s_json to get a more base-class interpretation of an input parameter.

Essentially, "obvious" (to me) means you don't need to know what the rest of the expression is/does, there's only one possible answer (which is the case for new/as/cast).

Reply:
Yeah, the foofy part I was taking advantage of here is that I wanted a shorter line for better reading comprehension. Felt it payed off more than not having to know what the parse method does. I'm not losing sleep over it one way or another. :)